narwhals/_arrow/series.py

```diff
@@ -1017,6 +1017,16 @@ def __iter__(self: Self) -> Iterator[Any]:
             for x in self._native_series.__iter__()
         )
 
+    def __contains__(self: Self, other: Any) -> bool:
+        import pyarrow as pa  # ignore-banned-imports
+        import pyarrow.compute as pc  # ignore-banned-imports
+
+        native_series = self._native_series
+        return maybe_extract_py_scalar(  # type: ignore[no-any-return]
+            pc.is_in(pa.scalar(other, type=native_series.type), native_series),
```

Comment:
if `other` is of a type not convertible to `native_series.type`, will this raise? I'm wondering if we should return `False` in such a case?

Reply:
looks like polars would also raise here:
```
In [2]: 'foo' in pl.Series([1,2,3])
---------------------------------------------------------------------------
InvalidOperationError                     Traceback (most recent call last)
Cell In[2], line 1
----> 1 'foo' in pl.Series([1,2,3])

File ~/scratch/.venv/lib/python3.12/site-packages/polars/series/series.py:1233, in Series.__contains__(self, item)
   1231 if item is None:
   1232     return self.has_nulls()
-> 1233 return self.implode().list.contains(item).item()

File ~/scratch/.venv/lib/python3.12/site-packages/polars/series/utils.py:106, in call_expr.<locals>.wrapper(self, *args, **kwargs)
    104     expr = getattr(expr, namespace)
    105 f = getattr(expr, func.__name__)
--> 106 return s.to_frame().select_seq(f(*args, **kwargs)).to_series()

File ~/scratch/.venv/lib/python3.12/site-packages/polars/dataframe/frame.py:9138, in DataFrame.select_seq(self, *exprs, **named_exprs)
   9115 def select_seq(
   9116     self, *exprs: IntoExpr | Iterable[IntoExpr], **named_exprs: IntoExpr
   9117 ) -> DataFrame:
   9118     """
   9119     Select columns from this DataFrame.
   9120
   (...)
   9136     select
   9137     """
-> 9138     return self.lazy().select_seq(*exprs, **named_exprs).collect(_eager=True)

File ~/scratch/.venv/lib/python3.12/site-packages/polars/lazyframe/frame.py:2029, in LazyFrame.collect(self, type_coercion, predicate_pushdown, projection_pushdown, simplify_expression, slice_pushdown, comm_subplan_elim, comm_subexpr_elim, cluster_with_columns, collapse_joins, no_optimization, streaming, engine, background, _eager, **_kwargs)
   2027 # Only for testing purposes
   2028 callback = _kwargs.get("post_opt_callback", callback)
-> 2029 return wrap_df(ldf.collect(callback))

InvalidOperationError: is_in operation not supported for dtypes `str` and `list[i64]`
```
probably ok to just follow them here then

Reply:
still, probably good to have a test to check it raises? we can leave the expression unification for a separate topic

Reply:
I will add a test and make sure that same exception is raised 👌

Reply:
Ok this is definitly wrong behaviour for arrow as the casting would convert from float to int, leading to the following:

```py
import pyarrow as pa
import narwhals as nw

data = [100, 200, None]
s = nw.from_native(pa.table({"a": data}), eager_only=True)["a"]

100.3 in s
True
```

On the Exception side, I would not be sure how to catch the error for pandas as:

```py
(pd.Series([1,2,3]) == "foo").any()
np.False_
```

Reply:
tbh we could just dispatch do `(s._compliant_series == other).any()` for all backends 😄 that would still be better than the status quo